velox/exec/trace/QueryDataWriter.h

```diff
@@ -28,7 +28,10 @@ namespace facebook::velox::exec::trace {
 /// file.
 class QueryDataWriter {
  public:
-  explicit QueryDataWriter(const std::string& path, memory::MemoryPool* pool);
+  explicit QueryDataWriter(
+      std::string path,
+      memory::MemoryPool* pool,
+      uint64_t maxTraceBytes);
```

Comment:
We might need a callback to make sure the traced input bytes from a task won't exceed limit like spill bytes tracking? Thanks!